### PR TITLE
feat: google auth, profile, logout, open, save, delete

### PR DESCRIPTION
**Pull request by rostyslavvnahornyi**
see: https://github.com/3DStreet/3dstreet-editor/issues/203

### CODE REVIEW

**Comment by kfarr** [comment]
thanks @rostyslavvnahornyi ~~when you get a chance can you also provide details and / or link to reference materials on how to provision the corresponding firebase app? I will test the current iteration of this PR using `threedstreets` firebase project ID as provided, but an immediate next step will be testing my ability to deploy again to another project ID.~~ Nevermind, figured it out!

I have been able to test auth with 3dstreet firebase config settings on localhost, yay! I have also made some minor updates to the text copy.

The last issue I see is that there appear to be many more files created in dist directory:
image

I'm curious why so many more now and are they all necessary? I ask because in some contexts we load the 3dstreet editor with 1 minified file and I'm not sure if it would be possible to specify multiple build files: https://github.com/3DStreet/3dstreet/blob/main/index.html#L47

**Comment by kfarr** [comment]
Some server setup notes:
- src/services/firebase.js file includes production server firebase config info. in theory it's ok to have this publicly committed but someday we may need private env keys. I have added domain restrictions to the public firebase apikey. Let me know if that causes issues during development.
- i added a custom domain for auth domain instead of firebaseapp.com default
- Added google auth using firebase: authentication > sign in method > provider > google

next step:
- profile page
- then save / load - may need design help

**Comment by rostyslavnahornyi** [comment]
Hi @kfarr !
I spent a lot of time on making a separate page with react, but so far it has not worked out. I need to pay more attention to this in the future, because the project will need it. But so far, unfortunately. I hope we talk about it tomorrow)

In general, I made the profile a modal window with logout and user info, without connecting the logic of the scenes - which will have to be done in the next task.

File: src/index.js [hunk 1]
```diff
@@ -72,7 +72,7 @@ Inspector.prototype = {
     div.setAttribute('data-aframe-inspector', 'app');
     document.body.appendChild(div);
     const root = createRoot(div);
-    root.render(<Main />);
```
Comment (kfarr) [inline src/index.js:75]: maybe we can put auth context here for now until router is needed?

File: src/components/App.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { Suspense, lazy } from 'react';
+import { BrowserRouter, Route, Routes } from 'react-router-dom';
+import { AuthProvider } from '../contexts';
+
+const MainPage = lazy(() => import('./Main'));
+
+const App = () => (
+  <AuthProvider>
```
Comment (kfarr) [inline src/components/App.js]: Not sure we need router, suspense, etc. right now.

it seems like authprovider is the only thing we need right now?

Maybe we can delete this file and just add authprovider in index.js, such as
```
<AuthProvider>
   <MainPage />
</AuthProvider>
```

File: package.json [hunk 3]
```diff
@@ -26,10 +26,12 @@
     "@types/uuid": "^9.0.0",
     "classnames": "^2.3.2",
     "clipboard": "^2.0.11",
+    "firebase": "^9.23.0",
     "lodash-es": "^4.17.21",
     "prop-types": "^15.8.1",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-router-dom": "^6.14.0",
```
Comment (kfarr) [inline package.json]: could we get rid of this dependency for now?

**Review by kfarr — COMMENTED** [review]
some questions / feedback @rostyslavvnahornyi

**Pull request closed without merging**